AK/Error.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Error.h>
+#include <AK/String.h>
+
+namespace AK {
+
+Error::Error(Error&& other)
+{
+    swap(this->m_syscall, other.m_syscall);
+    swap(this->m_code, other.m_code);
+#if defined(KERNEL)
+    swap(this->m_string_literal, other.m_string_literal);
+#else
+    if (other.m_string_literal.has<StringView>())
+        this->m_string_literal = other.m_string_literal.get<StringView>();
+    else if (other.m_string_literal.get<String const*>() != nullptr) {
+        this->m_string_literal = other.m_string_literal.get<String const*>();
+        // Make sure the other Error doesn't destruct the string!
+        other.m_string_literal.get<String const*>() = nullptr;
+    }
+#endif
+}
+
+Error::Error(Error const& other)
+    : m_code(other.m_code)
+    , m_syscall(other.m_syscall)
+{
+#if defined(KERNEL)
+    this->m_string_literal = other.m_string_literal;
+#else
+    if (other.m_string_literal.has<StringView>())
+        this->m_string_literal = other.m_string_literal.get<StringView>();
+    else if (other.m_string_literal.get<String const*>() != nullptr)
+        this->m_string_literal = new (nothrow) String { *other.m_string_literal.get<String const*>() };
```

Comment:
"and this empty error belongs to...someone"

Reply:
Why empty? We only create empty errors via static StringViews.